File: backend/src/SkillPilot.Domain/Enums/SkillLevel.cs
namespace SkillPilot.Domain.Enums;

/// <summary>
/// スキルの習熟レベル。<see cref="Entities.UserSkill"/>(ユーザーの保有レベル)と
/// <see cref="Entities.SkillResult"/>(求人が求めるレベル)の双方で共用する。
/// </summary>
public enum SkillLevel
{
    /// <summary>初級。</summary>
    Beginner,

    /// <summary>中級。</summary>
    Intermediate,

    /// <summary>上級。</summary>
    Advanced
}

File: frontend/src/lib/skillLevel.ts
import { SkillLevel } from "@/api/generated/models";

/** スキル習熟レベルの日本語表示。求人分析結果・プロフィールのスキル管理で共通利用する。 */
export const SKILL_LEVEL_LABEL: Record<SkillLevel, string> = {
  [SkillLevel.Beginner]: "初級",
  [SkillLevel.Intermediate]: "中級",
  [SkillLevel.Advanced]: "上級",
};
